src/Handlers/RedisHandler.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace CodeIgniter\Settings\Handlers;
+
+use CodeIgniter\I18n\Time;
+use Config\Cache;
+use RuntimeException;
+
+/**
+ * Provides cache persistence for Settings.
+ * Uses ArrayHandler for storage to minimize cache calls.
+ */
+class RedisHandler extends ArrayHandler
+{
+    /**
+     * The Query Builder for the Settings table.
+     */
+    private BaseRedisHandler $handler;
+
+    /**
+     * Array of contexts that have been stored.
+     *
+     * @var ?string[]
+     */
+    private $hydrated = [];
+
+    /**
+     * Array of contexts that have been stored.
+     */
+    private Cache $config;
+
+    /**
+     * Stores the configured database table.
+     */
+    public function __construct()
+    {
+        $this->config  = new Cache();
+        $this->handler = new BaseRedisHandler($this->config);
+        $this->handler->initialize();
+    }
+
+    /**
+     * Checks whether this handler has a value set.
+     */
+    public function has(string $class, string $property, ?string $context = null): bool
+    {
+        $this->hydrate($context);
+
+        return $this->hasStored($class, $property, $context);
+    }
+
+    /**
+     * Attempt to retrieve a value from the database.
+     * To boost performance, all of the values are
+     * read and stored the first call for each contexts
+     * and then retrieved from storage.
+     *
+     * @return mixed|null
```

Comment:
Please do not use @mixed as possible.
If it is really `@mixed`, `null` should be removed.